main/src/ca/uwaterloo/flix/language/phase/MonoTypes.scala

```diff
@@ -20,84 +20,21 @@ import ca.uwaterloo.flix.api.Flix
 import ca.uwaterloo.flix.language.ast.Ast.CaseSymUse
 import ca.uwaterloo.flix.language.ast.LoweredAst.{Expr, Pattern}
 import ca.uwaterloo.flix.language.ast.Type.eraseAliases
-import ca.uwaterloo.flix.language.ast.{Ast, AtomicOp, LoweredAst, Scheme, SourceLocation, Symbol, Type, TypeConstructor}
+import ca.uwaterloo.flix.language.ast.{Ast, AtomicOp, Kind, LoweredAst, Scheme, SourceLocation, Symbol, Type, TypeConstructor}
 import ca.uwaterloo.flix.language.phase.unification.{Substitution, TypeNormalization}
+import ca.uwaterloo.flix.util.collection.MapOps
 import ca.uwaterloo.flix.util.{InternalCompilerException, ParOps}
 
+import scala.annotation.tailrec
 import scala.collection.mutable
 
 /**
   * This phase does two things:
-  * - Specializes polymorphic enums with the specific types that it is used as.
+  * - Erase enums, such that `Option[t]` becomes `Option`
```

Comment:
I don't think this has to be a phase anymore. We can just roll it into MonoDefs. 

(Then later, separately, we can rename that back).

Reply:
This would be very nice. Then monomorph can be understood as just one phase.

Reply:
I agree, but can we make it a separate PR? I don't think its trivial where to 1) erase alias and 2) transform the types given that monoDefs doesn't even visit every type as it is

Reply:
OK